### ISSUE TITLE
greenThumbPerk has no effect

### ISSUE THREAD

**Issue by GarrixWong** [SteveTownsend/SmartHarvestSE#124]
Code in SHSE_EventsAlias.psc OnHarvest seems to be problematic.

```
        elseif baseForm.GetType() == getType_kFlora || baseForm.GetType() == getType_kTree

            ; "Flora" or "Tree" Producer REFRs cannot be identified by item type
            bool applyGreenThumb = (greenThumbPerk && player.HasPerk(greenThumbPerk)) as bool
            bool suppressMessage = silent || applyGreenThumb
            ;DebugTrace("Flora " + baseForm.GetName())
            if ActivateEx(akTarget, player, suppressMessage)
                ;we must send the message if required default would have been incorrect
                notify = !silent && applyGreenThumb
                if applyGreenThumb
                    float greenThumbValue = greenThumbPerk.GetNthEntryValue(0, 0)
                    int countPP = ((count * greenThumbValue) - count) as int
                    if (countPP >= 1)
                        player.AddItem(baseForm, countPP, true)
                        count = count + countPP
                    endif
                endIf
            endif
```
`if countPP >= 1`, the vm will throw an error like 'can't put object into container', because you can't add a flora or tree to actor's inventory. I fixed it by using `GetIngredient`, the code below works fine for me.
```
        elseif baseForm.GetType() == getType_kFlora || baseForm.GetType() == getType_kTree

            ; "Flora" or "Tree" Producer REFRs cannot be identified by item type
            Flora floraForm = baseForm as Flora
            if floraForm && floraForm.GetIngredient()
              baseForm = floraForm.GetIngredient()
            endif
            TreeObject treeForm = baseForm as TreeObject
            if treeForm && treeForm.GetIngredient()
              baseForm = treeForm.GetIngredient()
            endif
            bool applyGreenThumb = (greenThumbPerk && player.HasPerk(greenThumbPerk)) as bool
            bool suppressMessage = silent || applyGreenThumb
            ;DebugTrace("Flora " + baseForm.GetName())
            if ActivateEx(akTarget, player, suppressMessage)
                ;we must send the message if required default would have been incorrect
                notify = !silent && applyGreenThumb
                if applyGreenThumb
                    float greenThumbValue = greenThumbPerk.GetNthEntryValue(0, 0)
                    int countPP = ((count * greenThumbValue) - count) as int
                    if (countPP >= 1)
                        player.AddItem(baseForm, countPP, true)
                        count = count + countPP
                    endif
                endIf
            endif
```

**Comment by GarrixWong** [comment]
BTW, I use a perk overhaul and it has another perk like greenThumbPerk, so I changed the greenThumbPerk property to match the new perk.

**Comment by SteveTownsend** [comment]
thanks, I'll have another look at this. I did try this with the vanilla
Green Thumb, but maybe I misunderstood the results.

**Comment by SteveTownsend** [comment]
I guess I nuked the GreenThumb perk property at some point? my SHSE ESP does not even have a setting for `greenThumbPerk`. Anyway, before releasing I'd like to roll in whatever you had to do to make this work for you using that alternate related Perk. So please let me know.

**Comment by GarrixWong** [comment]
My mod modified the perk Alchemist40 [PERK:000C07CB] to give more ingredient.
I find that without these code my perk can also take effect. With these code enabled harvesting will give more ingredients than expect. So if your mod doesn't use it, you may delete these codes otherwise it will produce error log when harvest.

**Comment by SteveTownsend** [comment]
I am reworking this whole area to take the scripting logic mostly away - is
your mod published?

**Comment by GarrixWong** [comment]
My mod is SCSI-PerksOfSkyrim's donated version, It only had free version published. I downloaded it from a website like patreon.  So you don't need to adapt to that mod.

**Comment by SteveTownsend** [comment]
OK, I'll just go ahead with the Green Thumb handling. Thanks. Fixed for v1.8.0.0
